test/Microsoft.ML.Benchmarks/PredictionEngineBench.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using BenchmarkDotNet.Attributes;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Learners;
+
+namespace Microsoft.ML.Benchmarks
+{
+    [Config(typeof(PredictConfig))]
+    public class PredictionEngineBench
```

Comment:
Can this class be `internal` and `sealed`?

Reply:
cc @adamsitnik

Reply:
@markusweimer no, the benchmarking engine generates some boilerplate code and one of the thing it does it creates a new type that derives from the type with benchmarks. So the types must no be static, sealed or nonpublic,